tcpclient/connectionsManagement_test.go

```diff
@@ -9,11 +9,13 @@ import (
 	"io/ioutil"
 )
 
+// We really need to refactor this test. We should verify connections do become established,
```

Comment:
let's create an issue? (I will suggest a specific issue to create a complete set of tests)